client/java/src/main/java/io/openlineage/client/utils/DatasetIdentifier.java

```diff
@@ -41,6 +42,12 @@ public DatasetIdentifier withSymlink(Symlink symlink) {
     return this;
   }
 
+  public DatasetIdentifier merge(DatasetIdentifier ident) {
+    symlinks.add(new Symlink(ident.getName(), ident.getNamespace(), SymlinkType.UNKNOWN));
+    symlinks.addAll(ident.getSymlinks());
```

Comment:
What if there are common symlinks?

Reply:
Good catch - however, how would you determine that? IMO that's still on the caller of this method.

Reply:
What's on the caller of this method? I thought just about converting `List` into `Set`

Reply:
Then we need good equals and hashcode 🙂 
I think those are the questions we need to answer, list vs set is just a technicality

Reply:
I left `TABLE` for now, however, will revisit that later.

Reply:
I don't see there is this method used, and why `di1.merge(di2)` expects that d2 is always a TABLE symlink.